Allow `Always show dark maps` option on CarPlay. (#3882)
mapbox/mapbox-navigation-ios: CHANGELOG.md

### Routing

* Added `Router.finishRouting()` method to finish routing session without dismissing related UI and logic components. ([#3880](https://github.com/mapbox/mapbox-navigation-ios/pull/3880))

### Map

* Added the `layerPosition` parameter to the `NavigationMapView.show(_:layerPosition:legIndex:)` method for controlling the position of the main route layer while presenting routes. ([#3897](https://github.com/mapbox/mapbox-navigation-ios/pull/3897))

### CarPlay

> * Fixed an issue where shields would become invisible when users drive through the tunnel. ([#3882](https://github.com/mapbox/mapbox-navigation-ios/pull/3882))

Reviewer: ```suggestion
* Fixed an issue where route shields disappeared when the user enters a tunnel. ([#3882](https://github.com/mapbox/mapbox-navigation-ios/pull/3882))
```

> * Added the ability to change map style depending on the option that was set in `Always show dark maps`. ([#3882](https://github.com/mapbox/mapbox-navigation-ios/pull/3882))

Reviewer: This is under the v2.6.0 section, but the milestone is v2.5.0 – which release should it land in?

Author: Definitely `v2.6.0`, too late to land it in scope of `v2.5.0` because of potential issues. We'll have more time to test in scope of `v2.6.0`, changed milestone.

> * Added the ability to change map style depending on the option that was set in `Always show dark maps`. ([#3882](https://github.com/mapbox/mapbox-navigation-ios/pull/3882))

Reviewer: ```suggestion
* The map automatically chooses the night style when “Always Show Dark Maps” is enabled in the Appearance section of Settings. ([#3882](https://github.com/mapbox/mapbox-navigation-ios/pull/3882))
```
